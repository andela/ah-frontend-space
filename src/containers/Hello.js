/* eslint-disable no-shadow */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import propTypes from 'prop-types';
import FirstAction from '../actions/FirstAction';


export class Hello extends Component {
  componentDidMount() {
    const { FirstAction } = this.props;
    FirstAction();
  }

  render() {
    return (
      <div>
        <h1>wassap</h1>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  index: state.index,
});

Hello.propTypes = {
  FirstAction: propTypes.func.isRequired,
};

export default connect(
  mapStateToProps,
  { FirstAction },
)(Hello);
