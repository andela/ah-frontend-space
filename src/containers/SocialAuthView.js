import React from 'react';
import propTypes from 'prop-types';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import { withRouter } from 'react-router-dom';
import SocialAuthButtons from '../components/socialAuth';
import SocialAuthActions from '../actions/socialAuthActions';

export class SocialAuthView extends React.Component {
  componentWillReceiveProps(nextProps) {
    if (nextProps.socialAuth.isAuthenticated) {
      const { history } = this.props;
      history.push('/');
    }
  }

    handleResponseGoogle = (response) => {
      const { google } = this.props;
      return google(response.tokenObj.id_token);
    }

    handleResponseFacebook = (response) => {
      const { facebook } = this.props;
      return facebook(response.accessToken);
    }

    handleGoogleFailure = (response) => {
      toast.error(response.error, {
        autoclose: 8000,
        hideProgressBar: false,
      });
    }

    render() {
      return (
        <SocialAuthButtons
          onSuccess={this.handleResponseGoogle}
          onFailure={this.handleGoogleFailure}
          faceBookResponse={this.handleResponseFacebook}
        />
      );
    }
}
export const mapStateToProps = state => ({
  socialAuth: state.socialAuth,
});
SocialAuthView.propTypes = {
  google: propTypes.func.isRequired,
  facebook: propTypes.func.isRequired,
  history: propTypes.shape().isRequired,
  push: propTypes.func,
  socialAuth: propTypes.shape(),
  isAuthenticated: propTypes.bool,

};
SocialAuthView.defaultProps = {
  push: () => {},
  socialAuth: {},
  isAuthenticated: false,

};

export default withRouter(connect(mapStateToProps,
  {
    google: SocialAuthActions.googleSignup,
    facebook: SocialAuthActions.facebookSignup,
  })(SocialAuthView));
