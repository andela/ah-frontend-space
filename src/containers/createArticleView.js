import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import firebase from '../firebase/firebase.config';
import createArticleAction from '../actions/createArticleActions';
import CreateArticle from '../components/createArticle';
import { Authenticated } from '../routes/Authenticated';


export class CreateArticleView extends Component {
    state = {
      article: {
        title: '',
        description: '',
        body: '',
      },
      image: '',
      status: 0,
    };

    componentWillReceiveProps(nextProps) {
      if (nextProps.isValid) {
        const { history } = this.props;
        history.push(`/${nextProps.payload.article.slug}`);
      }
    }

    onChangeHandlerArticle = (event) => {
      const { article } = this.state;
      const newArticle = {
        ...article,
        [event.target.name]: event.target.value,
      };
      this.setState({
        article: newArticle,
      });
    }

    /* istanbul ignore next */
    handleImageUpload = (event) => {
      if (event.target.files[0]) {
        const file = event.target.files[0];
        const upload = firebase.storage().ref(`images/${file.name}`).put(file);
        upload.then(() => {
          firebase.storage().ref(`images/${file.name}`).getDownloadURL().then((url) => {
            this.setState({
              image: url,
            });
          });
        });
        upload.on('state_changed', (snapshot) => {
          const status = Math.round(
            (snapshot.bytesTransferred / snapshot.totalBytes) * 100,
          );
          this.setState({ status });
        });
      }
    }


    onSubmitHandler = (event) => {
      const { createArticles } = this.props;
      const { article: { title, description, body }, image } = this.state;
      event.preventDefault();
      const data = {
        title,
        description,
        body,
        image,
      };
      createArticles(data);
    }

    render() {
      const { article: { title, description, body }, status, image } = this.state;
      return (
        <div>
          <CreateArticle
            title={title}
            description={description}
            body={body}
            image={this.handleImageUpload}
            onChange={this.onChangeHandlerArticle}
            onSubmit={this.onSubmitHandler}
            status={status}
            display={image}
          />
        </div>
      );
    }
}
CreateArticleView.propTypes = {
  isValid: PropTypes.bool.isRequired,
  payload: PropTypes.shape().isRequired,
  history: PropTypes.shape().isRequired,
  createArticles: PropTypes.func.isRequired,
};

export const mapStateToProps = state => ({
  isValid: state.createReducer.isValid,
  payload: state.createReducer.payload,
});

export default connect(mapStateToProps,
  { createArticles: createArticleAction })(Authenticated(CreateArticleView));
