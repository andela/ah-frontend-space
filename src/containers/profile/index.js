/* eslint-disable react/prop-types */
/* eslint-disable no-shadow */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropType from 'prop-types';
import { fetchProfile } from '../../actions/profileActions';
import ProfileComponent from '../../components/profile/ProfileFetch';
import { Authenticated } from '../../routes/Authenticated';


export class Profile extends Component {
  componentDidMount() {
    const { fetchProfile } = this.props;
    fetchProfile();
  }

  render() {
    const { profile } = this.props;
    return (<ProfileComponent profile={profile} />);
  }
}

export const mapStateToProps = state => (
  {
    profile: state.profile,
  }
);

Profile.propType = {
  props: PropType.object.isRequired,
  fetchProfile: PropType.func.isRequired,
  profile: PropType.object.isRequired,
};

export default connect(mapStateToProps, { fetchProfile })(Authenticated(Profile));
