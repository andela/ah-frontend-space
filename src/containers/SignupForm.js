import React, { Component } from 'react';
import { connect } from 'react-redux';
import propTypes from 'prop-types';

/* Import Components */
import Input from '../components/common/Input';
import signupAction from '../actions/signupAction';
import '../assets/css/signup.scss';
import spinner from '../assets/images/spinner.gif';
import SocialAuthView from './SocialAuthView';


function inputcontent(type, id, placeholder, name, value, onChange) {
  return (
    <Input
      type={type}
      id={id}
      placeholder={placeholder}
      name={name}
      value={value}
      onChange={onChange}
    />
  );
}
export class SignupForm extends Component {
  /* This life cycle hook gets executed when the component mounts */

  constructor() {
    super();
    this.state = {
      username: '',
      email: '',
      password: '',
    };

    this.changeHandler = this.changeHandler.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.isValid) {
      const { history } = this.props;
      history.push('/signin');
    }
  }

  changeHandler = (e) => {
    const { name, value } = e.target;
    this.setState({
      [name]: value,
    });
  }

  handleSubmit = (e) => {
    const { signupAction } = this.props;
    e.preventDefault();
    // get our form data out of state
    const {
      username, email, password,
    } = this.state;
    const data = {
      username,
      email,
      password,
    };
    signupAction(data);
  }

  signupFormContent = (username, email, password, isSigningUp) => (
    <div className="row signuppage">
      <div className="col-sm-6">
        <h2 className="text">
          Authors Haven is here to create a community of like
          minded authors to foster inspiration and innovation
          by leveraging the modern web. Create your account
          here to join the community
        </h2>
      </div>
      <div className="col-sm-6 form-contents">
        <h3 className="form-header">Signup here</h3>
        <form className="container" onSubmit={this.handleSubmit}>
          { isSigningUp ? (
            <div className="col-sm-6">
              {' '}
              <img src={spinner} alt="loading...." />
            </div>
          ) : (
            <div>
              {inputcontent('text', 'uname', 'Username', 'username', `${username}`, this.changeHandler)}
              {inputcontent('email', 'email', 'Email', 'email', `${email}`, this.changeHandler)}
              {inputcontent('password', 'pass', 'Password', 'password', `${password}`, this.changeHandler)}
            </div>
          )
              }
          <div className="row signupbutton">
            <div className="col-xs-6">
              <Input type="submit" id="signup" value="signup" name="submit" />
            </div>
          </div>
          <h4 className="words">OR</h4>
          <div className="row social-buttons">
            <SocialAuthView />
          </div>
        </form>
      </div>
    </div>

  )

  render() {
    const {
      username, email, password,
    } = this.state;
    const { isSigningUp } = this.props;
    return (
      <React.Fragment>
        { this.signupFormContent(username, email, password, isSigningUp) }
      </React.Fragment>
    );
  }
}

export const mapStateToProps = (state) => {
  const { isSigningUp, isValid } = state.signupReducer;
  return {
    isSigningUp,
    isValid,
  };
};

SignupForm.propTypes = {
  isSigningUp: propTypes.bool.isRequired,
  signupAction: propTypes.func.isRequired,
  isValid: propTypes.bool,
  history: propTypes.shape().isRequired,
  push: propTypes.func,
};

SignupForm.defaultProps = {
  push: () => {},
  isValid: false,
};

export default connect(mapStateToProps, { signupAction })(SignupForm);
