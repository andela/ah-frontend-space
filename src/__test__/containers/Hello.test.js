import React from 'react';
import { shallow } from 'enzyme';
import { Hello, mapStateToProps } from '../../containers/Hello';

describe('hello', () => {
  it('should match snapshot', () => {
    const props = {
      FirstAction: jest.fn(),
    };
    const wrapper = shallow(<Hello {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should map state to props', () => {
    const state = {
      index: {
        index: {
          message: 'hey',
        },
      },
    };
    expect(mapStateToProps(state)).toEqual({
      index: {
        index: {
          message: 'hey',
        },
      },
    });
  });
});
