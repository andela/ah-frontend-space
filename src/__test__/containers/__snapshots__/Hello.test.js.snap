// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hello should match snapshot 1`] = `ShallowWrapper {}`;
